### DISCUSSION TITLE
Ente as an Unraid docker app

### DISCUSSION THREAD

**Discussion by picajoso** [ente-io/ente#3767] [Enhancements]
I've seen that you've got a [support document](https://help.ente.io/self-hosting/) to selfhost through docker ([another version](https://github.com/ente-io/ente/blob/main/server/docs/docker.md)), but I wonder if it would be possible to create an Unraid docker app to make it easy for Unraid users to enjoy the self-hosted version of Ente.

**Comment by Magneticdud** [comment]
It's going to be more difficult than just using the docker compose plugin as unRAID docker apps can only have a single application while this one requires four (minio/garage, postgres, Museum, web)

The instructions would be more difficult to follow as you would have to manually create a network and manually configure all those separate applications in rhem

**Reply by picajoso** [reply]
I see. Happens the same with LibrePhotos, that I was able to install and run outside of the Unraid app store. A step-by-step tutorial on how to do this for Unraid users would be great.

**Reply by Magneticdud** [reply]
On my unRAID server I used the docker compose plugin and copied the basic yml with some limited modifications (mainly replaced minio with garage, but that's a bit more complex to set at the beginning)

On about the web part, I build it manually using WSL2 (on my local Windows computer) then just copy them on static hosting like GitHub pages

**Reply by picajoso** [reply]
Sounds great :) Would you consider writing a little tutorial/step-by-step to be able to replicate this? I'm not an advanced Unraid user and I'd probably make a lot of mistakes trying to do that.

**Reply by Magneticdud** [reply]
I'm not that advanced 😅 

I did it step by step.

First I configured garage (S3 compatible server) and understood how it worked. It took one week as it doesn't have an admin page and so everything (users, buckets, permissions, cors) needs to be set via API or cli tool. Maybe minio is easier to set, but I had the feeling that garage is made for self hosters while minio is made for enterprises first. After many tests and being able to see upload and backup everything with winscp, I passed to the next step, postgres.

I had no experience at all with postgresql, only mariadb. It seems a bit more complex, especially for the way that I wanted to use it (10 different databases with 10 different users in the same instance). Also it's not like mariadb where you can just target :lts and forget about it, but you need to choose a version and check back when in 1-2 years needs to update. So I did a test instance with dummy data, tried to access with heidisql, tried to invoke pgdumpall via docker exec for backups (essential because the database holds the encryption keys for the images), tried to restore that backup, and so on.

Then I tried to build the web portal on my computer, and hosted the result somewhere.

When I was familiar with everything, I put the final piece, that was museum, which connects S3 (files) with the database.

All the hard work (encryption, automatic tagging, sorting, and so on) is done by the clients which is a very interesting approach, I like it very much

**Reply by picajoso** [reply]
I used S3 a long time ago but now I self-host everything I can, but I'll try to check if I can do this on my Unraid machine. If I do, I'll update this, although it seems a little too much right now for me. I've been testing serveral alternatives and I'm pretty happy with Immich. I discovered Ente and I've used the free hosted service to check it out but I wanted to see if I could use it on Unraid. 

Thank so much for your comments Magneticdud :)

**Reply by ados8** [reply]
There's many apps which require a front and back end and even database e.g. postgres that work on Unraid.
The components are different images and an Unraid template is used per component.
Docker compose on Unraid is janky, always has been, they've promised to improve things in future but templates is the official process.

**Reply by ados8** [reply]
+1 for Unraid templates or a minimum docker images 👍.
